pkg/plugins/gateway/algorithms/least_load.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2024 The Aibrix Team.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package routingalgorithms
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/vllm-project/aibrix/pkg/cache"
+	"github.com/vllm-project/aibrix/pkg/types"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+)
+
+var (
+	ErrorNoAvailablePod = fmt.Errorf("no pod available")
+)
+
+type leastLoadRouter struct {
+	provider       cache.LoadProvider
```

Comment:
if user specific least load algorithm, but with profiling disabled, could that be an issue here?`provider` and `cappedProvider`